src/main/java/com/google/devtools/build/lib/bazel/bzlmod/CompiledModuleFile.java

```diff
@@ -0,0 +1,142 @@
+    if (!starlarkFile.ok()) {
+      Event.replayEventsOn(eventHandler, starlarkFile.errors());
+      throw ExternalDepsException.withMessage(
+          Code.BAD_MODULE, "error parsing MODULE.bazel file for %s", moduleKey);
+    }
+    try {
+      ImmutableList<IncludeStatement> includeStatements = checkModuleFileSyntax(starlarkFile);
+      Module predeclaredEnv =
+          Module.withPredeclared(
+              starlarkSemantics, starlarkEnv.getStarlarkGlobals().getModuleToplevels());
+      Program program = Program.compileFile(starlarkFile, predeclaredEnv);
+      return new CompiledModuleFile(moduleFile, program, predeclaredEnv, includeStatements);
+    } catch (SyntaxError.Exception e) {
+      Event.replayEventsOn(eventHandler, e.errors());
+      throw ExternalDepsException.withMessage(
+          Code.BAD_MODULE, "syntax error in MODULE.bazel file for %s", moduleKey);
+    }
+  }
+
+  @VisibleForTesting
+  static ImmutableList<IncludeStatement> checkModuleFileSyntax(StarlarkFile starlarkFile)
+      throws SyntaxError.Exception {
+    var includeStatements = ImmutableList.<IncludeStatement>builder();
+    new DotBazelFileSyntaxChecker("MODULE.bazel files", /* canLoadBzl= */ false) {
+      @Override
+      public void visit(ExpressionStatement node) {
+        // We can assume this statement isn't nested in any block, since we don't allow
+        // `if`/`def`/`for` in MODULE.bazel.
+        if (node.getExpression() instanceof CallExpression call
+            && call.getFunction() instanceof Identifier id
+            && id.getName().equals(INCLUDE_IDENTIFIER)) {
+          // Found a top-level call to `include`!
+          if (call.getArguments().size() == 1
+              && call.getArguments().getFirst() instanceof Argument.Positional pos
+              && pos.getValue() instanceof StringLiteral str) {
+            includeStatements.add(new IncludeStatement(str.getValue(), call.getStartLocation()));
+            // We can stop going down this rabbit hole now.
+            return;
+          }
+          error(
+              node.getStartLocation(),
+              "the `include` directive MUST be called with exactly one positional argument that "
+                  + "is a string literal");
+          return;
+        }
+        super.visit(node);
+      }
+
+      @Override
+      public void visit(DotExpression node) {
+        visit(node.getObject());
+        if (!node.getField().getName().equals(INCLUDE_IDENTIFIER)) {
+          // This is fine: `whatever.include`
+          // (so `include` can be used as a tag class name)
+          visit(node.getField());
+        }
+      }
+
+      @Override
+      public void visit(Identifier node) {
+        if (node.getName().equals(INCLUDE_IDENTIFIER)) {
+          // If we somehow reach the `include` identifier but NOT as the other allowed cases above,
+          // cry foul.
+          error(
```

Comment:
Just noting that extension proxies called `include` will break with this. I think that's fine, but *technically* not backwards compatible for 7.2.0. But maybe that's okay?

Reply:
Could we just return from the visitor after the first assignment to `include`? That would be similar to how you can override global symbols in Starlark.

Reply:
that's a good idea -- basically removes the risk. done.